lib/dialogue-agent/execution_dialogue_agent.ts

```diff
@@ -382,6 +382,9 @@ export default class ExecutionDialogueAgent extends AbstractDialogueAgent<undefi
     }
 
     private _tryGetStoredVariable(type : Type, variable : string) : Ast.Value|null {
+        if (this._dlg.isAnonymous)
```

Comment:
it's probably fine to get variables already stored in anonymous mode?

Reply:
It's kind of weird though. Why would you want that?

Reply:
for convenience in case that we already know the value? I suppose it's unlikely to switch from unanonymous to anonymous

Reply:
I mean, you can log in to the "anonymous" user as if it was a real user, and set variables there. I'm just unclear why you would want to do that.